cluster_tools/cluster_tools/schedulers/kube.py

```diff
@@ -0,0 +1,262 @@
+                    },
+                }
+            )
+
+    def get_python_executable(self):
+        return self.job_resources.get("python_executable", "python")
+
+    def inner_submit(
+        self, cmdline: str, job_name: Optional[str] = None, job_count: int = 1, **_
+    ):
+        """Starts a Kubernetes pod that runs the specified shell command line."""
+
+        kubernetes_client = KubernetesClient()
+        self.ensure_kubernetes_namespace()
+        array_job_id = str(uuid4())
+
+        job_id_future = concurrent.futures.Future()
+        job_id_future.set_result(array_job_id)
+        job_id_futures = [job_id_future]
+        ranges = [(0, job_count)]
+
+        requested_resources = {
+            k: v
+            for k, v in self.job_resources.items()
+            if k in ("memory", "cpu") or k.startswith("hugepages-")
+        }
+        umaskline = (
+            f"umask {self.job_resources['umask']}; "
+            if "umask" in self.job_resources
+            else ""
+        )
+        stdout_path = self.format_log_file_path(
+            self.cfut_dir, f"{array_job_id}_$JOB_COMPLETION_INDEX"
+        )
+        stderr_path = self.format_log_file_path(
+            self.cfut_dir, f"{array_job_id}_$JOB_COMPLETION_INDEX", suffix=".stderr"
+        )
+
+        job_manifest = {
+            "apiVersion": "batch/v1",
+            "kind": "Job",
+            "metadata": {
+                "name": array_job_id,
+            },
+            "spec": {
+                "completionMode": "Indexed",
+                "completions": job_count,
+                "parallelism": job_count,
+                "ttlSecondsAfterFinished": 604800,  # 7 days
+                "template": {
+                    "metadata": {
+                        "annotations": {
+                            "cluster-tools.scalableminds.com/job_id": array_job_id,
+                            "cluster-tools.scalableminds.com/job_name": job_name,
+                        },
+                    },
+                    "spec": {
+                        "containers": [
+                            {
+                                "image": self.job_resources["image"],
+                                "imagePullPolicy": "IfNotPresent",
+                                "workingDir": os.path.abspath(os.path.curdir),
+                                "command": ["/bin/bash"],
+                                "name": "worker",
+                                "args": [
+                                    "-c",
+                                    f"{umaskline}{cmdline} 0 > >(tee -a {stdout_path}) 2> >(tee -a {stderr_path} >&2)",
+                                ],
+                                "env": [
+                                    {"name": name, "value": value}
+                                    for name, value in os.environ.items()
+                                    if name not in ("PWD", "OLDPWD")
```

Comment:
In typical unix systems, `PWD` is always there. We don't want to propagate them, because we set the working directory explicitly. Don't know where `OLDPWD` comes from 🤷 , but seems unnesseccary.